sdk/python/feast/feature_view_projection.py

```diff
@@ -57,20 +80,57 @@ def from_proto(proto: FeatureViewProjectionProto):
             features=[],
             join_key_map=dict(proto.join_key_map),
             desired_features=[],
+            timestamp_field=proto.timestamp_field or None,
+            date_partition_column=proto.date_partition_column or None,
+            created_timestamp_column=proto.created_timestamp_column or None,
+            batch_source=proto.batch_source or None,
         )
         for feature_column in proto.feature_columns:
             feature_view_projection.features.append(Field.from_proto(feature_column))
 
         return feature_view_projection
 
+    @staticmethod
+    def from_feature_view_definition(feature_view: "FeatureView"):
+        # TODO need to implement this for StreamFeatureViews
+        if getattr(feature_view, "batch_source", None):
+            return FeatureViewProjection(
+                name=feature_view.name,
+                name_alias=None,
+                features=feature_view.features,
+                desired_features=[],
+                timestamp_field=feature_view.batch_source.created_timestamp_column
+                or None,
+                created_timestamp_column=feature_view.batch_source.created_timestamp_column
+                or None,
+                date_partition_column=feature_view.batch_source.date_partition_column
+                or None,
+                batch_source=feature_view.batch_source or None,
+            )
+
     @staticmethod
     def from_definition(base_feature_view: "BaseFeatureView"):
-        return FeatureViewProjection(
-            name=base_feature_view.name,
-            name_alias=None,
-            features=base_feature_view.features,
-            desired_features=[],
-        )
+        if getattr(base_feature_view, "batch_source", None):
+            return FeatureViewProjection(
+                name=base_feature_view.name,
+                name_alias=None,
+                features=base_feature_view.features,
+                desired_features=[],
+                timestamp_field=base_feature_view.batch_source.created_timestamp_column  # type:ignore[attr-defined]
```

Comment:
This is admittedly a hack and I don't like it but this should be refactored in 1.0.0

My overall learning from this is that FeatureViews should be really derived from the same object with the same class parameters and make them optionally instantiated. 

This should be described more thoroughly for 1.0.0